tendermint/src/abci/event.rs

```diff
@@ -111,47 +166,50 @@ where
     }
 }
 
-/// A key-value pair describing an [`Event`].
-///
-/// Generic methods are provided for more ergonomic attribute construction, see
-/// [`Event::new`] for details.
-///
-/// [ABCI documentation](https://docs.tendermint.com/master/spec/abci/abci.html#events)
-#[derive(Clone, PartialEq, Eq, Debug, Serialize, Deserialize, Hash)]
-pub struct EventAttribute {
-    /// The event key.
-    #[serde(with = "serializers::allow_null")]
-    pub key: String,
-    /// The event value.
-    #[serde(with = "serializers::allow_null")]
-    pub value: String,
-    /// Whether Tendermint's indexer should index this event.
-    ///
-    /// **This field is nondeterministic**.
-    pub index: bool,
-}
-
 impl EventAttribute {
     /// Checks whether `&self` is equal to `other`, ignoring the `index` field.
     pub fn eq_ignoring_index(&self, other: &Self) -> bool {
-        self.key == other.key && self.value == other.value
+        match (self, other) {
+            (EventAttribute::V034(a), EventAttribute::V034(b)) => {
+                a.key == b.key && a.value == b.value
+            },
+            (EventAttribute::V038(a), EventAttribute::V038(b)) => {
+                a.key == b.key && a.value == b.value
+            },
+            (EventAttribute::V037(a), EventAttribute::V037(b)) => {
+                a.key == b.key && a.value == b.value
+            },
+            // Shouldn't happen, comparing event attributes from different versions
+            _ => false,
+        }
     }
 
     /// A variant of [`core::hash::Hash::hash`] that ignores the `index` field.
     pub fn hash_ignoring_index<H: core::hash::Hasher>(&self, state: &mut H) {
         use core::hash::Hash;
         // Call the `Hash` impl on the (k,v) tuple to avoid prefix collision issues.
-        (&self.key, &self.value).hash(state);
+        match self {
+            EventAttribute::V034(attr) => {
+                (&attr.key, &attr.value).hash(state);
+            },
+            EventAttribute::V037(attr) => {
+                (&attr.key, &attr.value).hash(state);
+            },
+            EventAttribute::V038(attr) => {
+                (&attr.key, &attr.value).hash(state);
+            },
+        }
     }
 }
 
 impl<K: Into<String>, V: Into<String>> From<(K, V, bool)> for EventAttribute {
     fn from((key, value, index): (K, V, bool)) -> Self {
-        EventAttribute {
+        // TODO: support all versions
```

Comment:
@romac How can I get the version currently used here?

Reply:
We cannot, so I think we can either leave this as-is or remove the instance altogether and leave it up to the call to build the right event attribute version.